pkg/virt-controller/watch/vm.go

```diff
@@ -2576,6 +2580,28 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 	return vm, syncErr, nil
 }
 
+func (c *VMController) patchVMIInterfacesMacAddress(interfaces []virtv1.Interface, vmi *virtv1.VirtualMachineInstance) error {
+	if vmi == nil || vmi.DeletionTimestamp != nil {
+		return nil
+	}
+
+	indexedIfaces := netvmispec.IndexInterfaceSpecByName(interfaces)
+	var patches []string
+	for i, vmiSpecIface := range vmi.Spec.Domain.Devices.Interfaces {
+		if vmIface, exist := indexedIfaces[vmiSpecIface.Name]; exist {
+			patches = append(patches, fmt.Sprintf(`{ "op": "replace", "path": "/spec/domain/devices/interfaces/%d/macAddress", "value":  %q }`, i, vmIface.MacAddress))
+		}
+	}
+
+	if len(patches) > 0 {
+		if _, err := c.clientset.VirtualMachineInstance(vmi.Namespace).Patch(context.Background(), vmi.Name, types.JSONPatchType, controller.GeneratePatchBytes(patches), &v1.PatchOptions{}); err != nil {
+			return err
+		}
```

Comment:
You can return err directly here. 

```suggestion
		_, err := c.clientset.VirtualMachineInstance(vmi.Namespace).Patch(context.Background(), vmi.Name, types.JSONPatchType, controller.GeneratePatchBytes(patches), &v1.PatchOptions{})
			return err
```